### PR TITLE
[HOTFIX] Modified code to fix the degrade in compaction performance

### PR DESCRIPTION
**Pull request by manishgupta88**
Problem
Compaction performance for 3.5 billion degraded by 16-20%

Analysis:
Code modification in RawResultIerator.java has caused the problem wherein few extra checks are performed as compared to previous code

Fix
Revert the changes in RawResultIerator class

 - [ ] Any interfaces changed?
 No
 - [ ] Any backward compatibility impacted?
 NA
 - [ ] Document update required?
No
 - [ ] Testing done
Verified in cluster testing       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6184/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7807/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6531/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6189/

File: core/src/main/java/org/apache/carbondata/core/scan/result/iterator/RawResultIterator.java [hunk 1]
```diff
@@ -53,153 +39,124 @@
...
-          }
-          // copy backup buffer to current buffer and fill backup buffer asyn
-          currentIdxInBuffer = 0;
-          currentBuffer = backupBuffer;
-          isBackupFilled = false;
-          fetchFuture = executorService.submit(new RowsFetcher(true));
-        } else {
-          currentIdxInBuffer = 0;
-          new RowsFetcher(false).call();
+  @Override public Object[] next() {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/scan/result/iterator/RawResultIterator.java]: Move @Override to previous line

Comment (manishgupta88) [reply core/src/main/java/org/apache/carbondata/core/scan/result/iterator/RawResultIterator.java]: ok

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6191/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7815/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6539/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6196/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6546/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7822/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6201/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7827/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6553/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**